openmls/src/treesync/node/leaf_node.rs

```diff
@@ -399,6 +400,41 @@ impl LeafNode {
         kp_key
     }
 
+    /// Generate a fresh leaf node.
+    ///
+    /// This includes generating a new encryption key pair that is stored in the
+    /// key store.
+    ///
+    /// This function can be used when generating an update. In most other cases
+    /// a leaf node should be generated as part of a new [`KeyPackage`].
+    pub fn generate(
```

Comment:
This function is meant to be part of the public API, right? Maybe it would be worth marking it somehow s.t. it won't be used internally in the future.